linera-views/src/memory.rs

```diff
@@ -43,11 +48,18 @@ pub const TEST_MEMORY_MAX_STREAM_QUERIES: usize = 10;
 /// The analog of the database is the BTreeMap
 pub type MemoryStoreMap = BTreeMap<Vec<u8>, Vec<u8>>;
 
+/// The container for the MemoryStopMap according to the Namespace.
+pub type NamespaceMemoryStore = BTreeMap<String, Arc<RwLock<MemoryStoreMap>>>;
+
+/// The global variables of the Namespace memory stores
+static MEMORY_STORES: Lazy<Mutex<NamespaceMemoryStore>> =
+    Lazy::new(|| Mutex::new(NamespaceMemoryStore::new()));
+
 /// A virtual DB client where data are persisted in memory.
 #[derive(Clone)]
 pub struct MemoryStore {
     /// The map used for storing the data.
-    pub map: Arc<RwLock<MutexGuardArc<MemoryStoreMap>>>,
+    pub map: Arc<RwLock<RwLockWriteGuardArc<MemoryStoreMap>>>,
```

Comment:
This looks suspicious to me because it looks like this `RwLockWriteGuardArc` is going to be held for a long time, so at runtime probably only one of these can exist at once — in which case it would make more sense to put the map directly here.  But I'm not sure I'm understanding the intent of this structure correctly.

Reply:
Ok, I implemented the change as requested. And yes, it works. Thank you.